File: 1037/1037.cpp
/* ************************************************************************** */
/*                                                                            */
/*                                                        :::      ::::::::   */
/*   1037.cpp                                           :+:      :+:    :+:   */
/*                                                    +:+ +:+         +:+     */
/*   By: jihoolee <jihoolee@student.42SEOUL.kr>     +#+  +:+       +#+        */
/*                                                +#+#+#+#+#+   +#+           */
/*   Created: 2022/05/09 12:45:18 by jihoolee          #+#    #+#             */
/*   Updated: 2022/05/09 13:38:06 by jihoolee         ###   ########.fr       */
/*                                                                            */
/* ************************************************************************** */

#include <iostream>
#include <set>

int main(void) {
  int           factor_num;
  int           factor;
  std::set<int> factors;

  std::cin >> factor_num;
  for (int i = 0; i < factor_num; ++i) {
    std::cin >> factor;
    factors.insert(factor);
  }
  std::cout << (*factors.begin()) * (*(--factors.end()));
  return 0;
}

File: 1037/1037.py
# **************************************************************************** #
#                                                                              #
#                                                         :::      ::::::::    #
#    1037.py                                            :+:      :+:    :+:    #
#                                                     +:+ +:+         +:+      #
#    By: jihoolee <jihoolee@student.42SEOUL.kr>     +#+  +:+       +#+         #
#                                                 +#+#+#+#+#+   +#+            #
#    Created: 2022/05/09 13:39:20 by jihoolee          #+#    #+#              #
#    Updated: 2022/05/09 13:44:17 by jihoolee         ###   ########.fr        #
#                                                                              #
# **************************************************************************** #

def main():
    _ = int(input())
    factors = list(map(int, input().split()))
    factors.sort()
    print(factors[0] * factors[-1])

if (__name__ == '__main__'):
    main()
